packages/oasis-actions/src/strategies/aave/adjust/adjust.ts

```diff
@@ -0,0 +1,729 @@
+import { getSwapDataHelper } from '../../../helpers/swap/getSwapData'
+import * as operations from '../../../operations'
+import { AAVEStrategyAddresses } from '../../../operations/aave/v2'
+import { AAVEV3StrategyAddresses } from '../../../operations/aave/v3'
+import { AaveProtocolData } from '../../../protocols/aave/getAaveProtocolData'
+import {
+  Address,
+  IOperation,
+  IPositionTransition,
+  IPositionTransitionArgs,
+  PositionType,
+  SwapData,
+} from '../../../types'
+import { AAVETokens } from '../../../types/aave'
+import { WithV2Addresses, WithV3Addresses } from '../../../types/aave/Addresses'
+import { WithFee } from '../../../types/aave/Fee'
+import { WithV2Protocol, WithV3Protocol } from '../../../types/aave/Protocol'
+import { getAaveTokenAddresses } from '../getAaveTokenAddresses'
+import { AaveVersion } from '../getCurrentPosition'
+
+export type AaveAdjustArgs = IPositionTransitionArgs<AAVETokens> & { positionType: PositionType }
+type AaveAdjustSharedDependencies = {
+  provider: providers.Provider
+  currentPosition: IPosition
+  getSwapData: (
+    fromToken: string,
+    toToken: string,
+    amount: BigNumber,
+    slippage: BigNumber,
+  ) => Promise<SwapData>
+  proxy: Address
+  user: Address
+  isDPMProxy: boolean
+}
+export type AaveV2AdjustDependencies = AaveAdjustSharedDependencies &
+  WithV2Addresses &
+  WithV2Protocol
+export type AaveV3AdjustDependencies = AaveAdjustSharedDependencies &
+  WithV3Addresses &
+  WithV3Protocol
+export type AaveAdjustDependencies = AaveV2AdjustDependencies | AaveV3AdjustDependencies
+
+export async function adjust(
+  args: AaveAdjustArgs,
+  dependencies: AaveAdjustDependencies,
+): Promise<IPositionTransition> {
+  if (isRiskIncreasing(dependencies.currentPosition.riskRatio, args.multiple)) {
+    return adjustRiskUp(args, dependencies)
+  } else {
+    return adjustRiskDown(args, dependencies)
+  }
+}
+
+async function adjustRiskUp(
+  args: AaveAdjustArgs,
+  dependencies: AaveAdjustDependencies,
+): Promise<IPositionTransition> {
+  const isAdjustUp = true
+  const fee = feeResolver(
+    args.collateralToken.symbol,
+    args.debtToken.symbol,
+    isAdjustUp,
+    args.positionType === 'Earn',
+  )
+
+  // Get quote swap
+  const estimatedSwapAmount = amountToWei(new BigNumber(1), args.debtToken.precision)
+  const { swapData: quoteSwapData } = await getSwapDataHelper<
+    typeof dependencies.addresses,
+    AAVETokens
+  >({
+    fromTokenIsDebt: true,
+    args: {
+      ...args,
+      fee,
+      swapAmountBeforeFees: estimatedSwapAmount,
+    },
+    addresses: dependencies.addresses,
+    services: {
+      getSwapData: dependencies.getSwapData,
+      getTokenAddresses: getAaveTokenAddresses,
+    },
+  })
+
+  // SimulateAdjustUp
+  const { simulatedPositionTransition: simulatedAdjustUp, reserveEModeCategory } =
+    await simulatePositionTransition(
+      isAdjustUp,
+      quoteSwapData,
+      { ...args, fee },
+      dependencies,
+      true,
```

Comment:
remove debug

Reply:
That's a flag for something else.

We could replace with named args to make more clear or just merge.